src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -74,9 +83,9 @@ client_ptr kafka_client_cache::fetch_or_insert(
                 // the timer is armed and it will run soon, then do nothing.
                 auto window = ss::lowres_clock::now() + 1s;
                 if (
-                  !_clean_and_evict_timer.armed()
-                  || (_clean_and_evict_timer.armed() && _clean_and_evict_timer.get_timeout() > window)) {
-                    _clean_and_evict_timer.rearm(window);
+                  !_gc_timer.armed()
+                  || (_gc_timer.armed() && _gc_timer.get_timeout() > window)) {
```

Comment:
seems complicated, maybe to simplify things another strategy could be:
1. let the timer fire every X seconds on interval
2. In the timer callback, check if some expiration has occurred:
  2a. If not enough time has passed since last expiry.... do nothing
  2b. Otherwise perform the cleanup logic

Reply:
The consequence of your suggestion is that all evicted clients are cleaned-up far into the future (The initial timer interval is 10s). This means that some resources may be left around longer than need be.

The point of this block of code is two fold:
1. Add some debounce logic so the eviction process triggers fewer times than necessary when there are multiple fibers querying the same cache
2. Eviction implies resources are freed soon if not immediately. So the time `window` is used to make sure those resources are freed sooner rather than later.

I'll keep this design the way it is for now.